src/XMakeBuildEngine/Evaluation/ConditionEvaluator.cs

```diff
@@ -64,52 +65,55 @@ internal static void UpdateConditionedPropertiesTable
                     bool lastPiece = pieceSeparator < 0;
                     int pieceEnd = lastPiece ? leftValue.Length : pieceSeparator;
 
-                    Match singlePropertyMatch = s_singlePropertyRegex.Match(leftValue, pieceStart, pieceEnd - pieceStart);
-
-                    if (singlePropertyMatch.Success)
+                    if (leftValue.StartsWith("$("))
```

Comment:
You're right, The regex should hopefully fail quickly in this case since it does not hit any fancy matchers for the starting characters. I'll revert it.